build/blog.ts

```diff
@@ -42,12 +46,24 @@ function calculateReadTime(copy: string): number {
 }
 
 async function readPost(
-  file: string
+  file: string,
+  extraMeta = true
 ): Promise<{ blogMeta: BlogPostFrontmatter; body: string }> {
   const raw = await fs.readFile(file, "utf-8");
 
   const { attributes, body } = frontmatter<BlogPostFrontmatter>(raw);
   const readTime = calculateReadTime(body);
+
+  if (extraMeta) {
+    const posts = await allPostFrontmatter();
+    const index = posts.findIndex((post) => post.slug === attributes.slug);
+    const filterFrontmatter = (
+      f?: BlogPostFrontmatter
+    ): BlogPostLimitedFrontmatter => f && { title: f.title, slug: f.slug };
+    attributes.previous = filterFrontmatter(posts[index + 1]);
+    attributes.next = filterFrontmatter(posts[index - 1]);
```

Comment:
🟢  I'm not a big fan of `readPost()` iterating - even indirectly - over all blog posts. Instead, consider extracting a `getPreviousAndNextPost(file: string)` function that takes the filename, looks in the list of all blog post filenames, and then specifically only reads the previous and next post (via the memoized `readRawPost()` function above). This better separates concerns, and also scales better as we add more blog posts.

PS: Same applies to content docs, where we would be better off with a similar approach, and differentiate `getDocument()` from `getDocumentMetadata()`, so that we can cache all documents' metadata, even if our LRU cache doesn't cache all documents' body.

Reply:
As our logic stands, our path names aren't prescriptive though: you could have a post at `foo/bar/foo/foobar-foobar/index.md`, and provided the frontmatter is well formed the post would render. If we are to rely on the path for sorting, then we'll need to enforce that the path matches the frontmatter - I'm not sure we want to do this.

Reply:
Afaik we already enforce in content / translated-content that the path matches the frontmatter slug, but I see the problem with this not being the case yet.